Installing-and-Configuring/OS/index.md

```diff
@@ -0,0 +1,106 @@
+---
+layout: default
+title: Operating Systems
+permalink: /OS/
+nav_order: 80
+parent: Installing and configuring
+has_children: yes
+has_toc: yes
+---
+
+<!-- markdownlint-disable MD033 -->
+<details open markdown="block">
+  <summary>
+    Table of contents
+  </summary>
+  {: .text-delta }
+1. TOC
+{:toc}
+</details>
+<!-- markdownlint-enable MD033 -->
+
+Heads and Operating Systems
+====
+
+Since a heads only system would be very limited you will need to install at least one Operating System.  The installation procedures vary greatly depending on which distro you choose, which version of that distro, and if you are encrypting partitions.  Some choices you will have to make include:
+
+* partition table
+* LUKS encryption (yes/no)
+* LUKS version (1 or 2)
+* single vs multiboot
+* simple distro vs hypervisor
+
+All together there are a lot of technical decisions.  This document will attempt to inform you enough to make the decisions.
+
+
+partition table
+----
+
+heads currently requires a separate boot partition which is a common configuration for Linux.  The heads model leaves the /boot partition unencrypted.  This is safe because heads will verify files at boot to detect tampering.
+
+When you partition storage make sure to create /boot or have the OS installer do it for you.  Generally, the boot partition is much smaller than the other OS partition(s).  You may keep everything else on 1 partition to make it easier to manage but the kernel and initrd on the boot partition can mount any number of others--encrypted or not.
+
+The other choices mentioned below may also affect the partition layout you choose.
+
+
+root partition encryption
+----
+
+If you have taken the trouble to install and use heads it is likely that you also want to use encryption to protect your data on storage devices.  LUKS is the most commonly used format in a heads based system.  
+
+Since the root partition is the end result of the boot process you may see how encrypting it makes the boot process complicated.  There are multiple schemes for making this setup work and more will probably be invented.  The key issue is deciding where to store the decryption key so that the root partition can be accessed during boot.  Most of the following use cases are out of scope for heads but worth a quick review so that you may make an informed decision.
+
+* key in TPM
+* key encrypted by security dongle (smartcard)
+* key on removable media
+
+For example, the [Pureboot](https://docs.puri.sm/PureBoot.html) model (based on heads) uses a security dongle (aka smartcard) to encrypt the LUKS decryption key and stores it on /boot which is not encrypted.  This key is decrypted on boot by the smartcard protected by a user PIN.  The smartcard is also used to sign the important files in /boot which are verified at each boot by heads.  
```

Comment:
For example, the [Pureboot](https://docs.puri.sm/PureBoot.html) model (based on heads) uses a security dongle (aka smartcard) to encrypt the LUKS decryption key and stores it on /boot which is not encrypted. 

That is not true. Please review https://docs.puri.sm/PureBoot/LibremKeyLUKS.html if you want to duplicate. But I would suggest to simply refer there for the user to follow the rabbit hole, this is out of scope of Heads and will lead to unmaintained doc.

Reply:
Given that Pureboot is the only commercial implementation of heads--that I know of--it is a good example to use as a reference in this doc where we point out the flexibility of Heads.  I do not have strong feelings about this and will remove if you strongly dislike.

perhaps some rewording might help...

"For example, the [Pureboot](https://docs.puri.sm/PureBoot.html) model (based on heads) uses a public key to encrypt a LUKS decryption key and store it on /boot which is not encrypted.  This key is decrypted on boot using the user PIN and the security dongle.  The public key is also used to sign the important files in /boot which are verified at each boot by heads.  "

I fought with PureBoot for half of 2020 and got to know the process.  Here is the reference script that Purism provides for this:  https://source.puri.sm/pureos/packages/smartcard-key-luks/-/blob/master/smartcard-key-luks

The script uses the public key to encrypt a disk unlock key.  It places the encrypted disk unlock key in the initrd which is placed in /boot which is unencrypted.